Modelica/Blocks/Logical.mo

```diff
@@ -973,6 +973,41 @@ parameter \"terminationText\".
 
 </html>"));
   end TerminateSimulation;
+
+  block AssertCondition "Trigger error (or warning) and print message if assertion condition is not fulfilled"
+
+    Modelica.Blocks.Interfaces.BooleanOutput condition=true
+      "Trigger error (or warning) and print message when condition becomes false"
+      annotation (Dialog, Placement(transformation(extent={{200,-10},{220,10}})));
+    parameter String message="... assertion condition no longer true"
+      "Text that will be displayed when assertion condition is false";
+    parameter AssertionLevel assertionLevel=AssertionLevel.error
+      "Select warning (1) or error (2)";
+  equation
+    assert(condition,message,assertionLevel);
+    annotation (Icon(coordinateSystem(
+          preserveAspectRatio=true,
```

Comment:
This has the same issue as #3986 and highlights one of the issues of duplicated code.
```suggestion
          preserveAspectRatio=false,
```